shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java

```diff
@@ -404,91 +415,247 @@ public void run() {
     }
   }
 
+  // Keep a queue of ImageReaders.
+  // Each ImageReader holds acquired Images.
+  // When we acquire the next image, close any ImageReaders that don't have any
+  // more pending images.
   @Keep
   @TargetApi(29)
   final class ImageReaderSurfaceProducer
       implements TextureRegistry.SurfaceProducer, TextureRegistry.ImageConsumer {
     private static final String TAG = "ImageReaderSurfaceProducer";
     private static final int MAX_IMAGES = 4;
 
+    private static final boolean VERBOSE_LOGS = false;
+
     private final long id;
 
     private boolean released;
     private boolean ignoringFence = false;
 
     private int requestedWidth = 0;
     private int requestedHeight = 0;
+    private boolean createNewReader = true;
```

Comment:
Could we add a 1-line comment for some of these, just so it's more clear why/what we're storing here?

Reply:
Added comments for all the fields.